ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -108,52 +110,60 @@
   private final Set<QueryMetadata> allLiveQueries;
   private final KsqlEngineMetrics engineMetrics;
   private final ScheduledExecutorService aggregateMetricsCollector;
+  private final Supplier<SchemaRegistryClient> schemaRegistryClientFactory;
   private final SchemaRegistryClient schemaRegistryClient;
   private final QueryIdGenerator queryIdGenerator;
   private final KafkaClientSupplier clientSupplier;
 
   public KsqlEngine(final KsqlConfig ksqlConfig) {
     this(
         new KafkaTopicClientImpl(ksqlConfig.getKsqlAdminClientConfigProps()),
-        new KsqlSchemaRegistryClientFactory(ksqlConfig).create(),
+        new KsqlSchemaRegistryClientFactory(ksqlConfig),
+        new KsqlSchemaRegistryClientFactory(ksqlConfig).get(),
         new DefaultKafkaClientSupplier(),
         new MetaStoreImpl(new InternalFunctionRegistry())
     );
   }
 
   public KsqlEngine(final KafkaTopicClient kafkaTopicClient,
-                    final SchemaRegistryClient schemaRegistryClient,
+                    final Supplier<SchemaRegistryClient> schemaRegistryClientFactory,
                     final KafkaClientSupplier clientSupplier
   ) {
     this(
         kafkaTopicClient,
-        schemaRegistryClient,
+        schemaRegistryClientFactory,
+        schemaRegistryClientFactory.get(),
         clientSupplier,
         new MetaStoreImpl(new InternalFunctionRegistry())
     );
   }
 
   // called externally by tests only
   public KsqlEngine(final KafkaTopicClient topicClient,
-                    final SchemaRegistryClient schemaRegistryClient,
+                    final Supplier<SchemaRegistryClient> schemaRegistryClientFactory,
                     final MetaStore metaStore
   ) {
     this(
         topicClient,
-        schemaRegistryClient,
+        schemaRegistryClientFactory,
+        schemaRegistryClientFactory.get(),
         new DefaultKafkaClientSupplier(),
         metaStore
     );
   }
 
   // called externally by tests only
   KsqlEngine(final KafkaTopicClient topicClient,
+             final Supplier<SchemaRegistryClient> schemaRegistryClientFactory,
```

Comment:
Is there a reason why we have both the `Supplier<SchemaReg...>` and the `SchemaRegistryClient`? Seems unnecessary?

Reply:
Nope. I'll clean it up.